### ISSUE TITLE
Can't install packages when upgraded the node from v8 to v14.15.4

### ISSUE THREAD

**Issue by slim-hmidi** [nodejs/node-gyp#2332]
* **Node Version**:  14.15.4 and 6.14.8
* **Platform**:   Windows_NT 10.0.17134
* **Compiler**: 
* **Module**:  install the package.json packages

<details><summary>Verbose output (from npm or node-gyp):</summary>

```
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\xxx\\AppData\\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\xxx\myProject\node_modules\mmmagic
gyp ERR! node -v v14.15.4
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
.
.
.

npm verb stack Error: mmmagic@0.4.5 install: `node-gyp rebuild`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (C:\Users\xxx\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
npm verb stack     at EventEmitter.emit (events.js:315:20)
npm verb stack     at ChildProcess.<anonymous> (C:\Users\xxx\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
npm verb stack     at ChildProcess.emit (events.js:315:20)
npm verb stack     at maybeClose (internal/child_process.js:1048:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
npm verb pkgid mmmagic@0.4.5
npm verb cwd C:\Users\x199377\AppData\Local\Programs\Git\Ego\ego\ego-app
npm verb Windows_NT 10.0.17134
npm verb argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "i" "--verbose"
npm verb node v14.15.4
npm verb npm  v6.14.8
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mmmagic@0.4.5 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mmmagic@0.4.5 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm verb exit [ 1, true ]
npm timing npm Completed in 83077ms

```

</details>

I followed the steps mentioned on the documentation and I installed python 3.8 and Visual C++ build tools.But I'm always getting that issue.

**Comment by cclauss** [comment]
> gyp ERR! node-gyp -v v5.1.0

Current is v7.1.2 so please upgrade.

**Comment by slim-hmidi** [comment]
should I upgraded npm? because I upgraded node-gyp but it does not fix the issue:
```
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@14.15.5 | win32 | x64
gyp info find Python using Python version 3.8.7 found at "C:\Users\xxx\AppData\Local\Programs\Python\Python38\python.exe"
gyp info find VS using VS2019 (16.8.31005.135) found at:
gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools"
gyp info find VS run with --verbose for detailed information
gyp info spawn C:\Users\xxx\AppData\Local\Programs\Python\Python38\python.exe
gyp info spawn args [
gyp info spawn args   'C:\\Users\\x199377\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\xxx\\AppData\\Local\\Programs\\Git\\Ego\\ego\\ego-app\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\xxx\\AppData\\Local\\node-gyp\\Cache\\14.15.5\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\xxx\\AppData\\Local\\node-gyp\\Cache\\14.15.5',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\xxx\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\14.15.5\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=C:\\Users\\xxx\\AppData\\Local\\Programs\\Git\\Ego\\ego\\ego-app',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\Users\\xxx\\AppData\\Local\\Programs\\Git\\Ego\\ego\\ego-app\\build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp: binding.gyp not found (cwd: C:\Users\xxx\AppData\Local\Programs\Git\Ego\ego\ego-app) while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Users\xxx\AppData\Roaming\npm\node_modules\node-gyp\lib\configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\xxx\AppData\Local\Programs\Git\Ego\ego\ego-app
gyp ERR! node -v v14.15.5
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok
```

**Comment by slim-hmidi** [comment]
I fixed it following @rvagg comment on #2272
